src/Layout/redist/targets/BundledSdks.targets

```diff
@@ -4,5 +4,7 @@
     <BundledSdk Include="Microsoft.NET.Sdk.WindowsDesktop" Version="$(MicrosoftNETSdkWindowsDesktopPackageVersion)" Condition="'$(DotNetBuildFromSource)' != 'true'" />
     <BundledSdk Include="FSharp.NET.Sdk" Version="1.0.4-bundled-0100" />
     <BundledSdk Include="Microsoft.Docker.Sdk" Version="1.1.0" />
+    <BundledSdk Include="Microsoft.NET.ILLink.Tasks" Version="$(MicrosoftNETILLinkTasksPackageVersion)" />
+    <BundledSdk Include="Microsoft.DotNet.ILCompiler" Version="$(MicrosoftDotNetILCompilerPackageVersion)" Condition="'$(NativeAotSupported)' != 'false'" />
```

Comment:
https://github.com/dotnet/sdk/pull/28690 removed this BundledSdk. We shouldn't add it back.
It invalidates https://github.com/dotnet/sdk/pull/28674.

I think we shouldn't include anything from https://github.com/dotnet/sdk/pull/28674, except maybe reserving `AotNotSupported/NETSDK1196` as as an identifer, and the updated text for `AotNoValidRuntimePackageError`.

We can do a separate PR against `main` to make PublishAot an optional feature for the Mono-based builds.

cc @sbomer @dsplaisted @agocke @crummel @omajid